test/metabase/api/dashboard_test.clj

```diff
@@ -1442,36 +1442,42 @@
-                                                   :row              3
-                                                   :dashboard_tab_id -1
-                                                   :card_id          card-id-2}]})]
+        (let [resp (mt/user-http-request :rasta :put 200 (format "dashboard/%d" dashboard-id)
+                                         {:name      "Updated dashboard name"
+                                          :tabs      [{:id   dashtab-id-1
+                                                       :name "Tab 1 edited"}
+                                                      {:id   dashtab-id-2
+                                                       :name "Tab 2"}
+                                                      {:id   -1
+                                                       :name "New tab"}]
+                                          :dashcards [{:id               dashcard-id-1
+                                                       :size_x           4
+                                                       :size_y           4
+                                                       :col              1
+                                                       :row              1
+                                                      ;; initialy was in tab1, now in tab 2
+                                                       :dashboard_tab_id dashtab-id-2
+                                                       :card_id          card-id-1}
+                                                      {:id               dashcard-id-2
+                                                       :dashboard_tab_id dashtab-id-2
+                                                       :size_x           2
+                                                       :size_y           2
+                                                       :col              2
+                                                       :row              2}
+                                                      ;; remove the dashcard3 and create a new card using negative numbers
+                                                      ;; and assign into the newly created dashcard
+                                                      {:id               -1
+                                                       :size_x           1
+                                                       :size_y           1
+                                                       :col              3
+                                                       :row              3
+                                                       :dashboard_tab_id -1
+                                                       :card_id          card-id-2}]})]
+          (testing "name got updated correctly"
+            (is (= "Updated dashboard name"
+                   (t2/select-one-fn :name :model/Dashboard :id dashboard-id)
+                   (:name resp))))
```

Comment:
This is the single new BE test. Every other change to this namespace just switches existing tests that use `PUT /api/dashboard/:id/cards` to use `PUT /api/dashboard/:id` instead.